### PR TITLE
Various improvements to django admin classes

### PR DESCRIPTION
**Pull request by gsidebo**
#### What are the relevant tickets?
No ticket. Noticed this while checking off RC boxes and such

#### What's this PR do?
Makes a bunch of improvements to Django admin classes to make them more usable and readable

#### How should this be manually tested?
(Required)

#### Where should the reviewer start?
Click through the affected Django admin classes, check out list pages, detail pages, and new object forms. Make sure everything looks correct

### CODE REVIEW

File: applications/admin.py [hunk 1]
```diff
@@ -166,43 +165,44 @@ def get_queryset(self, request):
...
+    get_run_display_title.short_description = "App Bootcamp Run"
+    get_run_display_title.admin_order_field = (
         "app_step_submission__bootcamp_application__bootcamp_run__title"
     )
 
     def app_step_submission_link(self, obj):
         """Returns a link to the related bootcamp application"""
-        app_step_submission = obj.app_step_submissions.first()
-        if app_step_submission is None:
+        app_step_submissions = obj.app_step_submissions.all()
```
Comment (annagav) [inline applications/admin.py:202]: why this change?

Comment (gsidebo) [reply applications/admin.py:202]: Takes advantage of the prefetched queryset. If you use `first()`, that runs an extra query.
